aws/rust-runtime/aws-sigv4/src/http_request/canonical_request.rs

```diff
@@ -28,6 +29,8 @@ pub(crate) mod header {
     pub(crate) const X_AMZ_DATE: &str = "x-amz-date";
     pub(crate) const X_AMZ_SECURITY_TOKEN: &str = "x-amz-security-token";
     pub(crate) const X_AMZ_USER_AGENT: &str = "x-amz-user-agent";
+    #[cfg(feature = "sigv4a")]
+    pub(crate) const X_AMZ_REGION_SET: &str = "x-amz-region-set";
```

Comment:
we can change this later, but I'd probably try to avoid needing to do line-by-line feature gating like this and instead isolate functionality into a module that can be gated when possible.